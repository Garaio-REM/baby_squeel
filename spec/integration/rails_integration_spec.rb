require 'spec_helper'

describe 'test that plain rails still works' do
  it 'joins and merge' do
    unless BabySqueel::ActiveRecord::VersionHelper.at_least_5_2_3?
      pending 'Rails does a LEFT OUTER JOIN for comments in the 5.2.0 version but an INNER JOIN in 5.2.3 +'
    end

    relation = Author.joins(:posts).merge(Post.joins(:comments).merge(Comment.where(body: 'body')))

    expect(relation).to match_sql_snapshot
  end

  it 'left_joins' do
    relation = Post.left_joins(:parent, :author)

    expect(relation).to match_sql_snapshot
  end
end
