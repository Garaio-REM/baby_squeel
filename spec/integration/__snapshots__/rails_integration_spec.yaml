---
test that plain rails still works joins and merge 1: SELECT "authors".* FROM "authors"
  INNER JOIN "posts" ON "posts"."author_id" = "authors"."id" INNER JOIN "comments"
  ON "comments"."post_id" = "posts"."id" WHERE "comments"."body" = 'body'
test that plain rails still works left_joins 1: SELECT "posts".* FROM "posts" LEFT
  OUTER JOIN "posts" "parents_posts" ON "parents_posts"."id" = "posts"."parent_id"
  LEFT OUTER JOIN "authors" ON "authors"."id" = "posts"."author_id"
